nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -189,7 +189,92 @@ in
       };
 
       buildMachines = mkOption {
-        type = types.listOf types.attrs;
+        type = types.loaOf (submodule ({ config }: {
+          options = {
+            sshUser = mkOption {
+              type = types.nullOr types.string;
+              default = null;
+              description = ''
+                The user as who to SSH to the build machine.
+              '';
+            };
+            hostName = mkOption {
+              type = types.string;
+              description = ''
+                The hostname of the build machine.
+              '';
+            };
+            system = mkOption {
+              type = types.string;
+              default = null;
+              description = ''
+                The system type the build machine can execute derivations on.
+                null if multiple are supported.
+              '';
+            };
+            systems = mkOption {
+              type = types.listOf types.string;
+              default = [];
+              description = ''
+                The system types the build machine can execute derivations on.
+              '';
+            };
+            sshKey = mkOption {
+              type = types.string;
+              default = "-";
```

Comment:
Because of https://github.com/NixOS/nixpkgs/issues/69900, it would be a good idea to force this option to not have a value from the store. This can be done with an `apply = toString`

Reply:
This isn't enough to guarantee this at all actually. See https://github.com/NixOS/nixpkgs/pull/78640 for a better way in the future. For now this should be just `types.str`